### PR TITLE
Added calculation implementation for simple length objects

### PR DESCRIPTION
**Pull request by njlawton**
Implemented calculations between simple lengths and produced tests to check there functionality.

Started outline of position-value.js

### CODE REVIEW

**Comment by wilddamon** [comment]
Please give the PR a more descriptive name, and make sure to add the chai and mocha files to your .gitignore.

**Comment by njlawton** [comment]
Changed name of pull request and removed node_modules folder

File: src/length-value.js [hunk 1]
```diff
@@ -44,19 +44,31 @@
   LengthValue.prototype = Object.create(shared.StyleValue.prototype);
 
   // Length Calculation Methods
-  LengthValue.prototype.add = function(value) {
-    throw new TypeError('Not implemented yet');
+  LengthValue.prototype.add = function(addedLength) {
+    if(this instanceof SimpleLength && addedLength instanceof SimpleLength){
+      if(this.type === addedLength.type){
+        return this.addSimpleLengths(addedLength);
+      }
```
Comment (wilddamon) [inline src/length-value.js]: You should collapse these two if statements into one, or add an else case to the inner one. At the moment it's possible to return undefined. Likewise below in subtract.

File: src/position-value.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(shared, scope, testing) {
+
+  /**
+   * PositionValue(xPos, yPos)
+   */
+  function PositionValue(xPos, yPos) {
```
Comment (wilddamon) [inline src/position-value.js]: Consider moving this into a separate pull request - it's not related to adding calculation functionality to SimpleLength.

File: src/simple-length.js [hunk 3]
```diff
@@ -38,7 +38,24 @@
...
+  SimpleLength.prototype.multiply = function(multiplier) {
+    return new SimpleLength((this.value * multiplier), this.type);
+  };
+
+  SimpleLength.prototype.divide = function(divider) {
+    return new SimpleLength((this.value / divider), this.type);
+  };
+
+  SimpleLength.prototype.addSimpleLengths = function(addedLength){
+    return new SimpleLength((this.value + addedLength.value), this.type);
```
Comment (wilddamon) [inline src/simple-length.js:55]: Please add some validation/sanity checks that the units are the same, or consider moving logic for (eventually) creating CalcLengths into this function.

**Comment by njlawton** [comment]
fixed issues stated above

File: src/simple-length.js [hunk 4]
```diff
@@ -38,7 +38,40 @@
...
+  SimpleLength.prototype._addSimpleLengths = function(addedLength){
+    if(this instanceof SimpleLength && addedLength instanceof SimpleLength && this.type === addedLength.type){
+      return new SimpleLength((this.value + addedLength.value), this.type);
+    } else {
+      if(!(this instanceof SimpleLength && addedLength instanceof SimpleLength)){
+        throw new TypeError('Objects not of type simple length');
+      } else {
+        throw new TypeError('SimpleLength units are not the same');
+      }
+    }
```
Comment (wilddamon) [inline src/simple-length.js:51]: The usual ordering for this is

if (error condition) {
// throw stuff
}

// do stuff

**Comment by njlawton** [comment]
Fixed

File: src/length-value.js [hunk 5]
```diff
@@ -44,19 +44,27 @@
   LengthValue.prototype = Object.create(shared.StyleValue.prototype);
 
   // Length Calculation Methods
-  LengthValue.prototype.add = function(value) {
-    throw new TypeError('Not implemented yet');
+  LengthValue.prototype.add = function(addedLength) {
+    if(this instanceof SimpleLength && addedLength instanceof SimpleLength && this.type === addedLength.type){
+      return this._addSimpleLengths(addedLength);
+    } else {
```
Comment (wilddamon) [inline src/length-value.js]: Don't need the else here since you are returning something in the if part.

File: src/simple-length.js [hunk 6]
```diff
@@ -38,7 +38,36 @@
...
+  SimpleLength.prototype.divide = function(divider) {
+    return new SimpleLength((this.value / divider), this.type);
+  };
+
+  SimpleLength.prototype._addSimpleLengths = function(addedLength){
+    if(!(this instanceof SimpleLength && addedLength instanceof SimpleLength && this.type === addedLength.type)){
+      if(!(this instanceof SimpleLength && addedLength instanceof SimpleLength)){
+        throw new TypeError('Objects not of type simple length');
+      }
+        throw new TypeError('SimpleLength units are not the same');
```
Comment (wilddamon) [inline src/simple-length.js]: Please fix indentation, also it might be clearer to separate these into two non-nested statements

if (not both SimpleLengths) {
// throw something
}
if (different units) {
// throw something
}

// do stuff

Same below.

**Comment by njlawton** [comment]
Solved git push problem

File: test/js/simple-length.js [hunk 7]
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
...
+    var simpleLen = new SimpleLength(5.3, 'px');
+    var calcOutput = simpleLen.multiply(3);
+    assert.strictEqual(simpleLen.type, 'px');
+    assert.strictEqual((5.3 * 3), calcOutput.value);
+  });
+ 
+  test('Division of a simple length produses a new simple length object', function() {
+    var simpleLen = new SimpleLength(27, 'px');
+    var calcOutput = simpleLen.divide(3);
+    assert.strictEqual(simpleLen.type, 'px');
```
Comment (nattridgell) [inline test/js/simple-length.js]: Was this meant to be checking calcOutput.type?

File: test/js/simple-length.js [hunk 8]
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
...
+  });
+ 
+  test('Division of a simple length produses a new simple length object', function() {
+    var simpleLen = new SimpleLength(27, 'px');
+    var calcOutput = simpleLen.divide(3);
+    assert.strictEqual(calcOutput.type, 'px');
+    assert.strictEqual(calcOutput.value, 9);
+  });
+
+  test('Division of a simple length that contains decimals produses correct output value', function() {
```
Comment (wilddamon) [inline test/js/simple-length.js]: s/produses/produces

File: test/js/simple-length.js [hunk 9]
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
...
   });
+
+  test('Multiplication of a simple length produses a new simple length object', function() {
+    var simpleLen = new SimpleLength(3, 'px');
+    var calcOutput = simpleLen.multiply(3);
+    assert.strictEqual(calcOutput.type, 'px');
+    assert.strictEqual(calcOutput.value, 9);
+  });
+
+  test('Multiplication of a simple length that contains decimals produses correct output value', function() {
```
Comment (wilddamon) [inline test/js/simple-length.js]: s/produses/produces

File: test/js/simple-length.js [hunk 10]
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
     assert.strictEqual(valueFromNumber.type, 'px');
     assert.strictEqual(valueFromNumber.value, 10);
   });
+
+  test('Multiplication of a simple length produses a new simple length object', function() {
```
Comment (wilddamon) [inline test/js/simple-length.js]: s/produses/produces
